src/compound/lens/IndexesLens.sol

```diff
@@ -115,44 +93,122 @@ abstract contract IndexesLens is LensStorage {
             indexes.poolSupplyIndex = ICToken(_poolToken).exchangeRateStored();
             indexes.poolBorrowIndex = ICToken(_poolToken).borrowIndex();
         } else {
-            (indexes.poolSupplyIndex, indexes.poolBorrowIndex) = getCurrentPoolIndexes(_poolToken);
+            (indexes.poolSupplyIndex, indexes.poolBorrowIndex, ) = _accruePoolInterests(
+                ICToken(_poolToken)
+            );
         }
 
-        if (!_updated || block.number == lastPoolIndexes.lastUpdateBlockNumber) {
-            indexes.p2pSupplyIndex = morpho.p2pSupplyIndex(_poolToken);
-            indexes.p2pBorrowIndex = morpho.p2pBorrowIndex(_poolToken);
+        (indexes.p2pSupplyIndex, indexes.p2pBorrowIndex, ) = _computeP2PIndexes(
+            _poolToken,
+            _updated,
+            indexes.poolSupplyIndex,
+            indexes.poolBorrowIndex,
+            delta,
+            lastPoolIndexes
+        );
+    }
+
+    /// @notice Returns the virtually updated pool indexes of a given market.
+    /// @dev Mimicks `CToken.accrueInterest`'s calculations, without writing to the storage.
+    /// @param _poolToken The address of the market.
+    /// @return poolSupplyIndex The supply index.
+    /// @return poolBorrowIndex The borrow index.
+    function _accruePoolInterests(ICToken _poolToken)
```

Comment:
(optional change) I'm not sure it's worth creating a new function here. Note that the third parameter of this function does not seem to be used (but I understand that this could help with stack too deep errors)

Reply:
Actually it is used to calculate future aave rates in `RatesLens`